index.js

```diff
@@ -20,302 +36,589 @@ var prefs = Cc["@mozilla.org/preferences-service;1"].getService(Ci.nsIPrefServic
+function getTickerConfigurationData(tickerId) {
+  var fontSize = getIntegerPreference("defaultFontSize");
+  if (fontSize <= 0) {
+    fontSize = DEFAULT_FONT_SIZE;
+  }
+  if (tickers[tickerId]) {
+    tickers[tickerId].enabled = getBooleanPreference("p" + tickerId);
+    tickers[tickerId].currencyPosition = getStringPreference("show-currency-label");
+    tickers[tickerId].color = getStringPreference("p" + tickerId + "Color");
+    tickers[tickerId].fontSize = fontSize;
+    tickers[tickerId].background = getBackgroundColor(tickerId);
+  }
+}
+
+/**
+ * Load tickers in the order that they are enabled in preferences
+ **/
+function loadDefaultTickers() {
+  for (var tickerId in tickers) {
+    if ( getBooleanPreference("p" + tickerId) ) { // Create Ticker
+      dlog("Loading ticker for " + tickerId);
+      updateTickerConfiguration(tickerId);
+      if (usingWidgets) {
+        createNewTickersWidget(tickerId);
       }
-    }
-    if (lowId.indexOf("litecoin") != -1) {
-      if (getBooleanPreference("silver-background")) { // Currency is Litecoin
-        return "silver";
+      if (tickers[tickerId].enabled) {
+        orderedTickers.push(tickerId);
       }
-    } else if (getBooleanPreference("gold-background")) { // Currency is Bitcoin
-      return "gold";
     }
-    return null;
   }
-
-  function getTickerConfigurationData(tickerId) {
-    var fontSize = getIntegerPreference("defaultFontSize");
-    if (fontSize <= 0) {
-      fontSize = DEFAULT_FONT_SIZE;
+  if ((orderedTickers != null) && (orderedTickers.length > 0)) {
+    storeTickersOrder();
+  }
+}
+
+/**
+ * Load the order of the tickers from Firefox's profile and create them
+ **/
+function loadTickersInOrder() {
+  var orderedActiveTickers = "";
+  try {
+    orderedActiveTickers = prefs.getCharPref("extensions.ADDON_ID.tickers_order");
+    if (orderedActiveTickers.length < 1) { // There is no order of tickers in addon set yet
+      loadDefaultTickers();
+      return;
     }
-    if (tickers[tickerId]) {
-      tickers[tickerId].enabled = getBooleanPreference("p" + tickerId);
-      tickers[tickerId].currencyPosition = getStringPreference("show-currency-label");
-      tickers[tickerId].color = getStringPreference("p" + tickerId + "Color");
-      tickers[tickerId].fontSize = fontSize;
-      tickers[tickerId].background = getBackgroundColor(tickerId);
+    var listOrderedTickers = orderedActiveTickers.split(",");
+    if (listOrderedTickers.length < 1) { // There is no order of tickers in addon set yet
+      loadDefaultTickers();
+      return;
     }
-  }
-
-  // Use tickers enabled in preferences to load in that order regardless of stored order
-  function loadDefaultTickers() {
-    for (var tickerId in tickers) {
-      if ( getBooleanPreference("p" + tickerId) ) { // Create Ticker
-        updateTickerConfiguration(tickerId);
-        if (tickers[tickerId].enabled) {
-          orderedTickers.push(tickerId);
-        }
+    for (var i in listOrderedTickers) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.